### PR TITLE
Update multiphysics examples

### PR DESCRIPTION
**Pull request by AmishgaAlphonius**
# Description of the problem

- Some multiphysics examples were outdated.

# Description of the solution

- This pull request is a first batch of updates on VOF examples

# How Has This Been Tested?

- All tests passed

# Documentation

- Documentation files of modified examples were updated. Here's a list of the updated examples:
- [x] Dam-Break
- [x] Rayleigh-Taylor Instability
- [x] Rising Bubble
- [x] Sloshing in a Rectangular Tank
- [x] Water Injection in a Closed Cell
- [x] Air Bubble Compression

# Future changes

- The rest of the multiphysics examples needs to be updated.

# Comments

- ``tests/dem/particle_point_contact.cc``: ``mpi_initialization`` maximum number of threads argument was changed to fix an "illegal instruction" error.
- An update for the "Static Bubble" example will be coming soon.

### CODE REVIEW

File: doc/source/examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability.rst [hunk 1]
```diff
@@ -64,15 +65,14 @@ Parameter File
...
-and the max CFL is :math:`0.9`.
+and the max CFL is :math:`0.8`.
 
 .. note::   
     This example uses an adaptive time-stepping method, where the 
-    time-step is modified during the simulation to keep the maximum value of the CFL condition below a given threshold (0.9 here). Using ``output control = time``, and ``output time = 0.005`` the simulation results are written every 0.005 s.
-
+    time-step is modified during the simulation to keep the maximum value of the CFL condition below a given threshold (:math:`0.9` here). Using ``output control = time``, and ``output time = 0.005`` the simulation results are written every :math:`0.005\, \text{s}`.
 
 .. code-block:: text
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability.rst:78]: At line 94 Replace : The ``multiphysics`` subsection enables to turn on ``true`` and off ``false`` the physics of interest. Here ``VOF`` and ``fluid dynamics`` are chosen (``fluid dynamics`` is true by default).

by : 

The ``multiphysics`` subsection enables to turn on (``true``) and off (``false``) the physics of interest. Here ``VOF`` and ``fluid dynamics`` are chosen (``fluid dynamics`` is ``true`` by default).

File: doc/source/examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability.rst [hunk 2]
```diff
@@ -329,35 +321,47 @@ In the following figure, we compare the simulation results with that of Garoosi
...
+|  .. image:: images/spike_and_bubble_evolution_ref_max_10_ref_min_8.png                |
+|     :alt: He et al comparison for a max refinement of 10 and a min refinement of 8.   |
+|           We see a better correspondence in the positions of the spike and the bubble.|
+|           However, for large values of t*, there is still gap between the positions.  |
+|     :align: center                                                                    |
+|     :width: 800                                                                       |
+|                                                                                       |
++---------------------------------------------------------------------------------------+
+
+The following figures shows the mass of ``fluid 1`` throughout the simulation with a constant (left) and adaptive (right) interface sharpening.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability.rst:356]:
```suggestion
The following figures show the mass of ``fluid 1`` throughout the simulation with a constant (left) and adaptive (right) interface sharpening.
```

File: doc/source/examples/multiphysics/rising-bubble/rising-bubble.rst [hunk 3]
```diff
@@ -240,32 +240,42 @@ To compare our simulation results to the literature, we extract the position and
...
-    :math:`\approx` 10 mins on 8 processes.
+    :math:`\sim \,7` minutes on :math:`8` processes.
 
 
 -----------------------
 Results and Discussion
 -----------------------
 
-The following image shows the shape and dimensions of the bubble after 3 seconds of simulation, and compares it with results of `[1] <https://doi.org/10.1002/fld.2643>`_, `[2] <https://doi.org/10.1002/fld.1934>`_.
+The following image shows the shape and dimensions of the bubble after :math:`3` seconds of simulation, and compares it with results of `[1] <https://doi.org/10.1002/fld.2643>`_, `[2] <https://doi.org/10.1002/fld.1934>`_.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/rising-bubble/rising-bubble.rst:263]:
```suggestion
The following image shows the shape and dimensions of the bubble after :math:`3` seconds of simulation, and compares it with results of `[1] <https://doi.org/10.1002/fld.2643>`_.
```
The comparison is from [1] only

File: doc/source/examples/multiphysics/sloshing-in-rectangular-tank/sloshing-in-rectangular-tank.rst [hunk 4]
```diff
@@ -50,7 +50,7 @@ In this problem, we simulate the damping of a small amplitude wave in a rectangu
 |                                                                                                                   |
 +-------------------------------------------------------------------------------------------------------------------+
 
-Four values of the Reynolds number are investigated: 2, 20, 200 and 2000. 
+Four values of the Reynolds number are investigated: :math:`2`, :math:`20`, :math:`200` and :math:`2000`.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/sloshing-in-rectangular-tank/sloshing-in-rectangular-tank.rst:53]: line 36 : replace " number values . Indeed," by " number values. Indeed,"

File: doc/source/examples/multiphysics/sloshing-in-rectangular-tank/sloshing-in-rectangular-tank.rst [hunk 5]
```diff
@@ -169,16 +170,17 @@ to run the simulation using eight CPU cores. Feel free to use more.
 .. warning:: 
     Make sure to compile lethe in `Release` mode and
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/sloshing-in-rectangular-tank/sloshing-in-rectangular-tank.rst:171]:
```suggestion
    Make sure to compile Lethe in `Release` mode and 
```

File: doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst [hunk 6]
```diff
@@ -61,7 +61,7 @@ The initial time step is set to :math:`0.005 \, \text{s}` and the simulation end
       set time end         = 0.49
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst:61]: line 25 :  replace  "Python script for postprocessing:" by "Postprocessing python script:" for uniformizing the notations with the other examples

File: doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst [hunk 7]
```diff
@@ -81,21 +81,22 @@ The ``multiphysics`` subsection is used to enable the VOF solver.
 VOF
 ~~~
 
-In the ``VOF`` subsection, the ``interface sharpening`` and the ``phase filtration`` features are enabled.
-The interface sharpening method and its parameters are explained in the :doc:`../dam-break/dam-break` example.
+In the ``VOF`` subsection, the ``compressible`` the ``interface sharpening`` and the ``phase filtration`` features are enabled.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst:84]:
```suggestion
In the ``VOF`` subsection, the ``compressible``, the ``interface sharpening``, and the ``phase filtration`` features are enabled.
```

File: doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst [hunk 8]
```diff
@@ -81,21 +81,22 @@ The ``multiphysics`` subsection is used to enable the VOF solver.
...
+The enabled ``compressible`` parameter allows interface compression by adding the :math:`\phi (\nabla \cdot \mathbf{u})` term to the VOF equation.
+The ``interface sharpening`` method and its parameters are explained in the :doc:`../dam-break/dam-break` example.
 The phase filtration filters the phase field used for the calculation of physical properties by stiffening the value of the phase fraction.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst:87]:
```suggestion
The ``phase filtration`` filters the phase field used for the calculation of physical properties by stiffening the value of the phase fraction.
```

File: doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst [hunk 9]
```diff
@@ -242,7 +243,7 @@ We can call the ``lethe-fluid`` by invoking the following command:
 to run the simulation using eight CPU cores.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst:243]: line 236 : remove the "the" before ``lethe-fluid``

**Review by PierreLaurentinCS — COMMENTED** [review]
Very good work for keeping the documentation updated. I pointed out some typos

**Review by blaisb — APPROVED** [review]
Well played, this is ready for a merge once @PierreLaurentinCS comment

File: doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst [hunk 10]
```diff
@@ -81,21 +81,22 @@ The ``multiphysics`` subsection is used to enable the VOF solver.
 VOF
 ~~~
 
-In the ``VOF`` subsection, the ``interface sharpening`` and the ``phase filtration`` features are enabled.
-The interface sharpening method and its parameters are explained in the :doc:`../dam-break/dam-break` example.
-The phase filtration filters the phase field used for the calculation of physical properties by stiffening the value of the phase fraction.
-We refer the reader to :doc:`../../../../theory/multiphysics/vof` theory guide for further explanation on the phase filtration.
+In the ``VOF`` subsection, the ``compressible``, the ``interface sharpening``, and the ``phase filtration`` features are enabled.
+The enabled ``compressible`` parameter allows interface compression by adding the :math:`\phi (\nabla \cdot \mathbf{u})` term to the VOF equation.
```
Comment (hepap) [inline doc/source/examples/multiphysics/water-injection-in-a-closed-cell/water-injection-in-a-closed-cell.rst:85]:
```suggestion
The enabled ``compressible`` parameter allows interface compression by adding the term :math:`\phi (\nabla \cdot \mathbf{u})` the VOF equation.
```

**Review by hepap — APPROVED** [review]
Not much to say! The copy button is nice :smile:

File: doc/source/examples/multiphysics/rising-bubble/rising-bubble.rst [hunk 11]
```diff
@@ -157,8 +157,8 @@ sufficiently smooth, as explained in the :doc:`../static-bubble/static-bubble` e
 Physical Properties
 ~~~~~~~~~~~~~~~~~~~~
 We define two fluids here simply by setting the number of fluids to be :math:`2`.
-In ``subsection fluid 0``, we set the density and the kinematic viscosity for the phase associated with a VOF indicator of 0. 
-A similar procedure is done for the phase associated with a VOF indicator of 1 in ``subsection fluid 1``. Then a ``fluid-fluid`` type of material interaction is added to specify the ``surface tension model``. In this case, it is set to ``constant`` with the ``surface tension coefficient`` :math:`\sigma` set to :math:`24.5`.
+In ``subsection fluid 0``, we set the density and the kinematic viscosity for the phase associated with a VOF indicator of :math:`0`.
+A similar procedure is done for the phase associated with a VOF indicator of :math:`1` in ``subsection fluid 1``. Then a ``fluid-fluid`` type of material interaction is added to specify the ``surface tension model``. In this case, it is set to ``constant`` with the ``surface tension coefficient`` :math:`\sigma` set to :math:`24.5`.
```
Comment (PierreLaurentinCS) [inline doc/source/examples/multiphysics/rising-bubble/rising-bubble.rst:161]:
```suggestion
A similar procedure is done for the phase associated with a VOF indicator of :math:`1` in ``subsection fluid 1``. Then a ``fluid-fluid`` type of ``material interaction`` is added to specify the ``surface tension model``. In this case, it is set to ``constant`` with the ``surface tension coefficient`` :math:`\sigma` set to :math:`24.5`.
```

**Review by PierreLaurentinCS — COMMENTED** [review]
Some remaining typos

**Review by PierreLaurentinCS — APPROVED** [review]

**Pull request merged**